src/main/java/jetbrains/buildServer/iaa/heuristics/Heuristic.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2000-2018 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
```

Comment:
Probably it could break encapsulation because one heuristics will have to know about another one.
@Linfar suggested me to provide the order of them not inside classes but in the spring config xml.
What do you think?